src/scaler/spreadscaler.rs

```diff
@@ -0,0 +1,1197 @@
+                                .iter()
+                                .map(|(_host_id, instances)| instances)
+                                .map(|instances| {
+                                    instances
+                                        .iter()
+                                        .filter(|instance| {
+                                            self.annotations(&spread.name).iter().all(
+                                                |(key, value)| {
+                                                    instance
+                                                        .annotations
+                                                        .get(key)
+                                                        .map(|v| v == value)
+                                                        .unwrap_or(false)
+                                                },
+                                            )
+                                        })
+                                        .collect::<Vec<&WadmActorInstance>>()
+                                })
+                                .map(|instance| instance.len())
+                                .sum()
+                        })
+                        .unwrap_or(0);
+
+                    match current_count.cmp(count) {
+                        Ordering::Equal => None,
+                        // Start actors to reach desired replicas
+                        Ordering::Less => Some(Command::StartActor(StartActor {
+                            reference: self.config.actor_reference.to_owned(),
+                            host_id: first_host.id.to_owned(),
+                            count: count - current_count,
+                            model_name: self.config.model_name.to_owned(),
+                            annotations: self.annotations(&spread.name),
+                        })),
+                        // Stop actors to reach desired replicas
+                        Ordering::Greater => Some(Command::StopActor(StopActor {
+                            actor_id: actor_id.to_owned(),
+                            host_id: first_host.id.to_owned(),
+                            count: current_count - count,
+                            model_name: self.config.model_name.to_owned(),
+                            annotations: self.annotations(&spread.name),
+                        })),
+                    }
+                } else {
+                    // No hosts were eligible, so we can't attempt to add or remove actors
+                    // We will want to return an error to indicate an unsatisfied Scaler here, eventually
+                    None
+                }
+            })
+            .collect::<Vec<Command>>();
+
+        Ok(commands)
+    }
+}
+
+impl<S: ReadStore + Send + Sync> ActorSpreadScaler<S> {
+    #[allow(unused)]
+    /// Construct a new ActorSpreadScaler with specified configuration values
+    fn new(
```

Comment:
Can you just `pub` this and remove the `allow(unused)` above?

Reply:
This doesn't seem to resolve the warning 🤔

Reply:
Is `ActorSpreadScaler` publicly exported in the `scaler` module?